service/src/main/java/tds/assessment/services/impl/AssessmentSegmentLoaderServiceImpl.java

```diff
@@ -0,0 +1,334 @@
+                int fieldTestMaxItemCount = 0;
+
+                for (Segment segment : assessment.getSegments()) {
+                    final SegmentBlueprintElement segmentBpElement = segment.segmentBlueprint().stream()
+                        .filter(bp -> bp.getIdRef().equals(segment.getId())) // get the actual segment blueprint (not any child claims/targets)
+                        .findFirst()
+                        .orElseThrow(() ->
+                            new TestPackageLoaderException("No matching segment blueprint element found in the test package for id {}", segment.getId()));
+
+                    minItemCount += segmentBpElement.getMinExamItems();
+                    maxItemCount += segmentBpElement.getMinExamItems();
+                    fieldTestMinItemCount += segmentBpElement.minFieldTestItems();
+                    fieldTestMaxItemCount += segmentBpElement.maxFieldTestItems();
+
+                    // Create a map to access the properties by name. Lowercasing the key to prevent casing issues
+                    final Map<String, String> itemSelectionProperties = segmentBpElement.itemSelection().stream()
+                        .collect(Collectors.toMap(p -> p.getName().toLowerCase(), Property::getValue));
+
+                    adminSubjects.add(new TblAdminSubject.Builder()
+                        .withKey(segment.getKey())
+                        .withClientName(testPackage.getPublisher())
+                        .withSubjectKey(subjectKey)
+                        .withId(segment.getId())
+                        .withStartAbility(Float.parseFloat(itemSelectionProperties.getOrDefault("startability", DEFAULT_START_ABILITY)))
+                        .withStartInfo(Float.parseFloat(itemSelectionProperties.getOrDefault("startinfo", DEFAULT_START_INFO)))
+                        .withMinItems(segmentBpElement.getMinExamItems())
+                        .withMaxItems(segmentBpElement.getMaxExamItems())
+                        .withSlope(Float.parseFloat(itemSelectionProperties.getOrDefault("slope", DEFAULT_SLOPE)))
+                        .withIntercept(Float.parseFloat(itemSelectionProperties.getOrDefault("intercept", DEFAULT_INTERCEPT)))
+                        .withFieldTestStartPos(itemSelectionProperties.containsKey("ftstartpos")
+                            ? Integer.parseInt(itemSelectionProperties.get("ftstartpos"))
+                            : null)
+                        .withFieldTestEndPos(itemSelectionProperties.containsKey("ftendpos")
+                            ? Integer.parseInt(itemSelectionProperties.get("ftendpos"))
+                            : null)
+                        .withFieldTestMinItems(segmentBpElement.minFieldTestItems())
+                        .withFieldTestMaxItems(segmentBpElement.maxFieldTestItems())
+                        .withSelectionAlgorithm(segment.getAlgorithmType().equals("adaptive") ? Algorithm.ADAPTIVE_2.getType() : segment.getAlgorithmType())
```

Comment:
Disregard my previous comments about using `.contains("adaptive")` - this answers my question(s) about using that as opposed to the `ADAPTIVE_2` enum value.

Reply:
Disregard my comments to your comments :)